drivers/memc/memc_mcux_flexspi_is66wvq8m4.c

```diff
@@ -0,0 +1,249 @@
+	},
+	/* Write Data (continuous burst) */
+	[WRITE_DATA] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_DDR, kFLEXSPI_4PAD, 0x22,
+				kFLEXSPI_Command_DDR, kFLEXSPI_4PAD, 0x00),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_RADDR_DDR, kFLEXSPI_4PAD, 16,
+				kFLEXSPI_Command_CADDR_DDR, kFLEXSPI_4PAD, 16),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_DUMMY_DDR, kFLEXSPI_4PAD, 28,
+				kFLEXSPI_Command_WRITE_DDR, kFLEXSPI_4PAD, 0x01),
+	},
+	/* Read Register */
+	[READ_REG] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_DDR, kFLEXSPI_4PAD, 0xCC,
+				kFLEXSPI_Command_DDR, kFLEXSPI_4PAD, 0x00),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_RADDR_DDR, kFLEXSPI_4PAD, 16,
+				kFLEXSPI_Command_CADDR_DDR, kFLEXSPI_4PAD, 16),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_DUMMY_DDR, kFLEXSPI_4PAD, 12,
+				kFLEXSPI_Command_READ_DDR, kFLEXSPI_4PAD, 0x01),
+	},
+	/* Write Register */
+	[WRITE_REG] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_DDR, kFLEXSPI_4PAD, 0x66,
+				kFLEXSPI_Command_DDR, kFLEXSPI_4PAD, 0x00),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_RADDR_DDR, kFLEXSPI_4PAD, 16,
+				kFLEXSPI_Command_CADDR_DDR, kFLEXSPI_4PAD, 16),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_WRITE_DDR, kFLEXSPI_4PAD, 0x01,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0x00),
+	},
+	/* Read Identification register */
+	[READ_ID] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_4PAD, 0xE0,
+				kFLEXSPI_Command_RADDR_DDR, kFLEXSPI_4PAD, 16),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_CADDR_DDR, kFLEXSPI_4PAD, 16,
+				kFLEXSPI_Command_DUMMY_RWDS_DDR, kFLEXSPI_4PAD, 0x08),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_READ_DDR, kFLEXSPI_4PAD, 0x01,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0x00),
+	}
+};
+
+
+/* Read vendor ID from identification register */
+static int memc_flexspi_is66wvq8m4_get_vendor_id(const struct device *dev,
+						uint8_t *vendor_id)
+{
+	const struct memc_flexspi_is66wvq8m4_config *config = dev->config;
+	struct memc_flexspi_is66wvq8m4_data *data = dev->data;
+	uint32_t buffer = 0;
+	int ret;
+
+	flexspi_transfer_t transfer = {
+		.deviceAddress = 0x00, /* Not used by this command */
+		.port = config->port,
+		.cmdType = kFLEXSPI_Read,
+		.SeqNumber = 1,
+		.seqIndex = READ_ID,
+		.data = &buffer,
+		.dataSize = 2,
+	};
+
+	ret = memc_flexspi_transfer(data->controller, &transfer);
+	*vendor_id = buffer & 0x7;
+
+	return ret;
+}
+
+/* Update configuration register */
+static int memc_flexspi_is66wvq8m4_update_cfg(const struct device *dev,
+					      uint16_t mask, uint16_t set_val)
+{
+	const struct memc_flexspi_is66wvq8m4_config *config = dev->config;
+	struct memc_flexspi_is66wvq8m4_data *data = dev->data;
+	uint32_t buffer = 0;
+	int ret;
+
+	flexspi_transfer_t transfer = {
+		/* Results in 0x4 being written on clock 4 */
+		.deviceAddress = (0x4 << 9),
+		.port = config->port,
+		.cmdType = kFLEXSPI_Read,
+		.SeqNumber = 1,
+		.seqIndex = READ_REG,
+		.data = &buffer,
+		.dataSize = 2,
+	};
+
+	ret = memc_flexspi_transfer(data->controller, &transfer);
+	if (ret < 0) {
+		return ret;
+	}
+
+	buffer &= (~mask  & 0xFFFF);
```

Comment:
Please consider GENMASK and FIELD_PREP().

Reply:
Switched to `GENMASK`, thanks for the suggestion. Not sure how I'd use `FIELD_PREP` here though? My understanding is that `FIELD_PREP(0xEF,0x1F0)` would expand to `0xF0`, since the macro will shift and mask the field. In this function though, `mask` is the bitmask that should be cleared from the buffer, and `set` has already been shifted to the correct location (so `FIELD_PREP(set, mask)`) wouldn't make sense, right?